cpp.md

```diff
@@ -0,0 +1,163 @@
+# C++ conventions
+
+### Compilation
+
+The table below contains the specifics of environment to build the projects. These settings are recommended and please make sure your project and machine matches these specs.
+
+| Topic                    | Config                                              |
+|--------------------------|-----------------------------------------------------|
+| C++ version              | C++ 14                                              |
+| Compiler                 | GCC / CLANG (11.0 +)                                |
+| Platforms                | iOS, android, amazon                                |
+| Build tools              | CMake (3.6+), gradle for android (5.x), pbx for iOS |
+| Compiler flags           | Warnings and Errors                                 |
+| Optional libs (external) | Boost                                               |
+
+We are also using CPP check to avoid any false positives in the project. You can easily install it on your system and learn more about it from - http://cppcheck.sourceforge.net/
+
+### Coding Style
+
+**Class name -** use camel case with first upper. eg
+``` C++
+class Person
+{
+  // ..
+};
+```
+
+**local variable name -**  use snake type. eg
+``` C++
+std::string first_name = “sameer”
+```
+
+**local class variables -** should be camel case with first letter lower case, eg
```

Comment:
camel case by definition means first letter is lower. the other one is upper camel case
https://en.wikipedia.org/wiki/Camel_case